Bug: 无法发送 MessageReference 消息段，JSON 序列化错误

断点发现尝试POST的数据中存在MessageReference，这个类（继承自BaseModel）在json序列化时抛异常

![Image](https://github.com/user-attachments/assets/500be0bb-e0fa-498e-8b81-d409886a3bb0)

报错日志：

```
01-23 01:23:03 [SUCCESS] nonebot | QQ xxxx| [EventType.READY]: {'event_id': None, 'version': 1, 'session_id': 'xxxx', 'user': {'id': 'xxxx', 'username': 'kuraku_bot-测试中', 'avatar': None, 'bot': True, 'union_openid': None, 'union_user_account': None}, 'shard': (0, 1)}
01-23 01:23:05 [SUCCESS] nonebot | QQ xxxx| [EventType.C2C_MESSAGE_CREATE]: Message ROBOT1.0_.xxxxxx!! from xxxxxx: [Text(type='text', data={'text': '/雀魂信息 无铭金重'})]
01-23 01:23:06 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot_plugin_majsoul.paifuya.query_majsoul_info, lineno=84)
01-23 01:23:07 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_majsoul.paifuya.query_majsoul_info, lineno=84) running complete
01-23 01:23:07 [ERROR] nonebot | Running Matcher(type='message', module=nonebot_plugin_majsoul.paifuya.query_majsoul_info, lineno=84) failed.
Traceback (most recent call last):
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\bot.py", line 566, in _request
    response = await self.adapter.request(request)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\internal\adapter\adapter.py", line 92, in request
    return await self.driver.request(setup)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\drivers\httpx.py", line 134, in request
    return await session.request(setup)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\drivers\httpx.py", line 78, in request
    response = await self.client.request(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\httpx\_client.py", line 1527, in request
    request = self.build_request(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\httpx\_client.py", line 378, in build_request
    return Request(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\httpx\_models.py", line 408, in __init__
    headers, stream = encode_request(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\httpx\_content.py", line 216, in encode_request
    return encode_json(json)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\httpx\_content.py", line 177, in encode_json
    body = json_dumps(
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type MessageReference is not JSON serializable

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\huang\PycharmProjects\bot-kuraku\bot\bot.py", line 23, in <module>
    nonebot.run(app="__mp_main__:app")
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\__init__.py", line 337, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\uvicorn\main.py", line 579, in run
    server.run()
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\windows_events.py", line 322, in run_forever
    super().run_forever()
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 641, in run_forever
    self._run_once()
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 1986, in _run_once
    handle._run()
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\utils.py", line 254, in run_coro_with_shield
    return await coro
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\message.py", line 506, in check_and_run_matcher
    await _run_matcher(
> File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\message.py", line 458, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 926, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 863, in simple_run
    await handler(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\dependencies\__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "C:\Users\huang\PycharmProjects\bot-kuraku\plugins\ssttkkl-nonebot-utils\ssttkkl_nonebot_utils\interceptor\handle_error.py", line 27, in wrapper
    return await func(*args, **kwargs)
  File "C:\Users\huang\PycharmProjects\bot-kuraku\plugins\ssttkkl-nonebot-utils\ssttkkl_nonebot_utils\interceptor\with_handling_reaction.py", line 20, in wrapper
    await func(*args, **kwargs)
  File "C:\Users\huang\PycharmProjects\bot-kuraku\plugins\nonebot-plugin-majsoul\src\nonebot_plugin_majsoul\paifuya\query_majsoul_info.py", line 77, in majsoul_info
    await wait_for(coro, timeout=conf.majsoul_query_timeout)
  File "C:\Users\huang\AppData\Local\Programs\Python\Python312\Lib\asyncio\tasks.py", line 520, in wait_for
    return await fut
  File "C:\Users\huang\PycharmProjects\bot-kuraku\plugins\nonebot-plugin-majsoul\src\nonebot_plugin_majsoul\paifuya\query_majsoul_info.py", line 192, in handle_majsoul_info
    await MessageFactory(Text(msg)).send(reply=True)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot_plugin_saa\abstract_factories.py", line 392, in send
    return await self._do_send(bot, target, event, at_sender, reply)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot_plugin_saa\abstract_factories.py", line 456, in _do_send
    return await sender(bot, self, target, event, at_sender, reply)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot_plugin_saa\adapters\qq.py", line 188, in send
    msg_return = await bot.send(event, message)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\bot.py", line 499, in send
    return await self.send_to_c2c(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\bot.py", line 416, in send_to_c2c
    return await self.post_c2c_messages(
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\internal\adapter\bot.py", line 176, in call_api
    raise exception
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\internal\adapter\bot.py", line 128, in call_api
    result = await self.adapter._call_api(self, api, **data)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\adapter.py", line 605, in _call_api
    return await api_handler(bot, **data)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\utils.py", line 54, in __call__
    return await self.func(inst, *args, **kwds)
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\bot.py", line 1691, in post_c2c_messages
    return type_validate_python(PostC2CMessagesReturn, await self._request(request))
  File "C:\Users\huang\AppData\Local\pypoetry\Cache\virtualenvs\bot-kuraku-bot-K0w-UOdi-py3.12\Lib\site-packages\nonebot\adapters\qq\bot.py", line 568, in _request
    raise NetworkError("API request failed") from e
nonebot.adapters.qq.exception.NetworkError: <NetWorkError message=API request failed>
```

--- a/nonebot/adapters/qq/bot.py
+++ b/nonebot/adapters/qq/bot.py
@@ -1680,7 +1680,11 @@ class Bot(BaseBot):
                         embed.dict(exclude_none=True) if embed is not None else None
                     ),
                     "image": image,
-                    "message_reference": message_reference,
+                    "message_reference": (
+                        message_reference.dict(exclude_none=True)
+                        if message_reference is not None
+                        else None
+                    ),
                     "event_id": event_id,
                     "msg_id": msg_id,
                     "msg_seq": msg_seq,
@@ -1782,7 +1786,11 @@ class Bot(BaseBot):
                         embed.dict(exclude_none=True) if embed is not None else None
                     ),
                     "image": image,
-                    "message_reference": message_reference,
+                    "message_reference": (
+                        message_reference.dict(exclude_none=True)
+                        if message_reference is not None
+                        else None
+                    ),
                     "event_id": event_id,
                     "msg_id": msg_id,
                     "msg_seq": msg_seq,

